### PR TITLE
Use unbounded channels in default p2p backend

### PR DESCRIPTION
**Pull request by pavel-kokolemin**
Using unbounded channels should fix the problem with potential deadlocks in p2p (when two channels are full).
I have removed a lot of async annotations that are no longer needed.

I think to apply back-pressure on the connected peers we could do something like this: count in `p2p::net::default_backend::peer::Peer::run` how many unprocessed requests  there are and if more than some limit then stop reading from the network socket, until some responses have been processed and sent back. I think this should solve the node overload problem nicely. It may look like this:

```
response = self.cmd_rx.recv() {
    // Request was processed, send response to the peer...
    request_count -= 1;
},
// Read from the socket if only there is some free capacity
request = self.socket.recv(), if request_count < 10 => {
    // New request received, send it to backend...
    request_count += 1;
}
```

We will need to ensure that for each request received, the backend would send a response. But I suppose it should be easy enough to do.

### CODE REVIEW

File: p2p/src/net/default_backend/backend.rs [hunk 1]
```diff
@@ -166,26 +169,27 @@ where
...
-        }
+
+        let peer = self
+            .peers
+            .remove(peer_id)
+            .ok_or(P2pError::PeerError(PeerError::PeerDoesntExist))?;
+
+        peer.tx.send(Event::Disconnect).map_err(P2pError::from)?;
+
+        Ok(())
```
Comment (iljakuklic) [inline p2p/src/net/default_backend/backend.rs:188]: Could it be just this?

```suggestion
        peer.tx.send(Event::Disconnect).map_err(P2pError::from)
```

There are quite a few occurrences of this pattern all over the place but not sure my proposal is much better. Not a big difference.

**Review by iljakuklic — APPROVED** [review]
Nice to see a bunch of `async`/`await` go away

**Review by stanislav-tkach — APPROVED** [review]

**Comment by TheQuantumPhysicist** [comment]
I'm not fully convinced this is needed. We should talk about this. Let's have a discussion about it tomorrow.

File: p2p/src/net/default_backend/backend.rs (hunk 1 above)
Comment (pavel-kokolemin) [reply p2p/src/net/default_backend/backend.rs]: Sure, updated, thanks.

**Review by TheQuantumPhysicist — APPROVED** [review]

**Pull request merged**